drivers/gles2/rasterizer_storage_gles2.cpp

```diff
@@ -682,7 +682,7 @@ void RasterizerStorageGLES2::texture_set_data(RID p_texture, const Ref<Image> &p
 		if (img == p_image) {
 			img = img->duplicate();
 		}
-		img->resize_to_po2(false);
+		img->resize_to_po2(false, Image::INTERPOLATE_NEAREST);
```

Comment:
Sorry for flip flopping. I changed my mind again. This should switch between nearest and linear based on the texture filter flag. 

```suggestion
		img->resize_to_po2(false, texture->flags & VS::TEXTURE_FLAG_FILTER ? Image::INTERPOLATE_BILINEAR : Image::INTERPOLATE_NEAREST);
```

Reply:
Done!